### PR TITLE
Fix #1. Pull request

### LINKED ISSUE #1: Impossible to resize Class/Interface/Structure box using Top Left handles
Top Left, Left and Top handles can't be used to resize the diagram element:
resize class

User should have possibility to resize the element using all visible handles.

### PR DESCRIPTION
**Pull request by mrpetro**
This fixes issue with resizing using top left shape handles. Snapping shapes to other shapes was also slightly improved.

### CODE REVIEW

**Comment by gbaychev** [comment]
Hey, I want just to let you know, that I saw your PR - thank you. I just don't know whether I'm gonna have the time look at it closely this week, will try to do that as soon as possible.

File: .gitignore [hunk 1]
```diff
@@ -214,4 +214,6 @@ FakesAssemblies/
 **/*.DesktopClient/ModelManifest.xml
 **/*.Server/GeneratedArtifacts
 **/*.Server/ModelManifest.xml
-_Pvt_Extensions
\ No newline at end of file
+_Pvt_Extensions
+/src/DiagramEditor/ClassDiagram/Shapes/PackageShape.cs
```
Comment (gbaychev) [inline .gitignore]: Please revert the .gitignore. It seems like you've added some WIP here

File: src/DiagramEditor/Diagrams/Shapes/ResizeEventArgs.cs [hunk 2]
```diff
@@ -21,19 +21,27 @@ namespace NClass.DiagramEditor.Diagrams.Shapes
...
-		public ResizeEventArgs(Size change)
-		{
-			this.change = change;
-		}
+        public ResizeEventArgs(SizeF positionChange, SizeF sizeChange)
+        {
+            this.positionChange = positionChange;
+            this.sizeChange = sizeChange;
+        }
+
```
Comment (gbaychev) [inline src/DiagramEditor/Diagrams/Shapes/ResizeEventArgs.cs]: Please add a ctor here, who sets positionChange = SizeF.Empty

File: src/DiagramEditor/Diagrams/Shapes/Shape.cs [hunk 3]
```diff
@@ -173,10 +171,8 @@ public virtual Size Size
 				if (size != value)
 				{
 					Size change = new Size(value.Width - Width, value.Height - Height);
-					if (IsResizing) //TODO: ez kell?
-						mouseDownLocation += change;
 					size = value;
-					OnResize(new ResizeEventArgs(change));
+					OnResize(new ResizeEventArgs(Size.Empty, change));
```
Comment (gbaychev) [inline src/DiagramEditor/Diagrams/Shapes/Shape.cs]: Use a different ctor here, please. See my notes about ResizeEventArgs.cs

File: src/DiagramEditor/Diagrams/Shapes/Shape.cs [hunk 4]
```diff
@@ -196,10 +192,8 @@ public virtual int Width
 				if (size.Width != value)
 				{
 					Size change = new Size(value - Width, 0);
-					if (IsResizing) //TODO: ez kell?
-						mouseDownLocation.X += change.Width;
 					size.Width = value;
-					OnResize(new ResizeEventArgs(change));
+					OnResize(new ResizeEventArgs(Size.Empty, change));
```
Comment (gbaychev) [inline src/DiagramEditor/Diagrams/Shapes/Shape.cs]: Use a different ctor here, please. See my notes about ResizeEventArgs.cs

File: src/DiagramEditor/Diagrams/Shapes/Shape.cs [hunk 5]
```diff
@@ -219,10 +213,8 @@ public virtual int Height
 				if (size.Height != value)
 				{
 					Size change = new Size(0, value - Height);
-					if (IsResizing) //TODO: ez kell?
-						mouseDownLocation.Y += change.Height;
 					size.Height = value;
-					OnResize(new ResizeEventArgs(change));
+					OnResize(new ResizeEventArgs(Size.Empty, change));
```
Comment (gbaychev) [inline src/DiagramEditor/Diagrams/Shapes/Shape.cs]: Use a different ctor here, please. See my notes about ResizeEventArgs.cs

**Comment by mrpetro** [comment]
Closing bad pull request due to new one.

**Pull request closed without merging**